internal/stores/autopilot.go

```diff
@@ -42,9 +35,45 @@ func (s *EphemeralAutopilotStore) SetConfig(c autopilot.Config) error {
 	return nil
 }
 
+// State implements autopilot.Store.
+func (s *EphemeralAutopilotStore) State() autopilot.State {
+	s.mu.Lock()
+	defer s.mu.Unlock()
+	return s.state
+}
+
+// SetState implements autopilot.Store.
+func (s *EphemeralAutopilotStore) SetState(state autopilot.State) error {
+	s.mu.Lock()
+	defer s.mu.Unlock()
+	s.state = state
+	return nil
+}
+
 // ProcessConsensusChange implements chain.Subscriber.
 func (s *EphemeralAutopilotStore) ProcessConsensusChange(cc modules.ConsensusChange) {
-	panic("unimplemented")
+	s.mu.Lock()
+	defer s.mu.Unlock()
+
+	for _, block := range cc.RevertedBlocks {
+		if block.ID() != types.GenesisID {
+			s.state.BlockHeight--
+		}
+	}
+	for _, block := range cc.AppliedBlocks {
+		if block.ID() != types.GenesisID {
+			s.state.BlockHeight++
+		}
+	}
+
+	// update current period
+	if s.state.BlockHeight >= s.state.CurrentPeriod+s.config.Contracts.Period {
+		s.state.CurrentPeriod += s.config.Contracts.Period
+	}
+
+	// update synced state
+	// TODO: might need a channel here
+	s.state.Synced = cc.Synced
```

Comment:
I don't think there is a point in persisting `Synced` as part of the state? Cause the consensus set should be able to tell us more accurately.

Reply:
Hm yeah I considered that, I didn't care for the time being because I want to use a channel anyway, `skyd` persisted the `Synced` state but I too found that a little odd. I even clear it out when I load it.

Reply:
this is chain state, so it should be tracked by `bus` rather than `autopilot`

Reply:
ok I'll move it 👍